keras/metrics/accuracy_metrics.py

```diff
@@ -113,11 +113,14 @@ class BinaryAccuracy(reduction_metrics.MeanMetricWrapper):
     ```
     """
 
-    def __init__(self, name="binary_accuracy", dtype=None):
-        super().__init__(fn=binary_accuracy, name=name, dtype=dtype)
+    def __init__(self, name="binary_accuracy", dtype=None, threshold=0.5):
+        if threshold is not None and (threshold <= 0 or threshold >= 1):
+            raise ValueError(f"`threshold` must be in interval (0, 1), but found: {threshold}")
```

Comment:
Standardize the message:

```
Invalid value for argument `threshold`. Expected a value in internal [0, 1]. Received: threshold={threshold}
```

Reply:
The notation `[0, 1]` typically represent a closed interval (0 and 1 included).
0 and 1 are invalid values for `threshold`, so I'd rather use the open interval notation `(0, 1)`. I've changed the message to: 
```python
f"Invalid value for argument `threshold`. Expected a value in internal (0, 1). Received: threshold={threshold}"
```